src/libs/kata-sys-util/src/mount.rs

```diff
@@ -0,0 +1,1029 @@
+
+        // - Normal mount without MS_REMOUNT flag
+        // - In the case of remounting with changed data (data != ""), need to call mount
+        if (flags & MsFlags::MS_REMOUNT) == MsFlags::empty() || !data.is_empty() {
+            mount_at(
+                chdir,
+                &self.source,
+                target.clone(),
+                &self.fs_type,
+                o_flag,
+                &data,
+            )?;
+        }
+
+        // Change mount propagation type.
+        if (flags & propagation_types) != MsFlags::empty() {
+            let propagation_flag = propagation_types | MsFlags::MS_REC | MsFlags::MS_SILENT;
+            mount(
+                Some(""),
+                &target,
+                Some(""),
+                flags & propagation_flag,
+                Some(""),
+            )
+            .map_err(|e| Error::Mount(PathBuf::new(), target.clone(), e))?;
+        }
+
+        // Bind mount readonly.
+        let bro_flag = MsFlags::MS_BIND | MsFlags::MS_RDONLY;
+        if (o_flag & bro_flag) == bro_flag {
+            do_remount_read_only(target, o_flag)?;
+        }
+
+        Ok(())
+    }
+}
+
+#[inline]
+fn do_remount_read_only<P: AsRef<Path>>(path: P, flags: MsFlags) -> Result<()> {
+    mount(
+        Some(""),
+        path.as_ref(),
+        Some(""),
+        flags | MsFlags::MS_BIND | MsFlags::MS_REMOUNT | MsFlags::MS_RDONLY,
+        Some(""),
+    )
+    .map_err(|e| Error::Remount(path.as_ref().to_path_buf(), e))
+}
+
+/// Take fstab style mount options and parses them for use with a standard mount() syscall.
+fn parse_mount_options(options: &[String]) -> Result<(MsFlags, String)> {
+    let mut flags: MsFlags = MsFlags::empty();
+    let mut data: Vec<String> = Vec::new();
+
+    for opt in options.iter() {
+        if opt == "defaults" {
+            continue;
+        } else if opt == "loop" {
+            return Err(Error::InvalidMountOption("loop".to_string()));
+        } else if let Some(v) = parse_mount_flags(flags, opt) {
+            flags = v;
+        } else {
+            data.push(opt.clone());
+        }
+    }
+
+    let data = data.join(",");
+    if data.len() > *MAX_MOUNT_PARAM_SIZE {
+        return Err(Error::MountOptionTooBig);
+    }
+
+    Ok((flags, data))
+}
+
+fn parse_mount_flags(mut flags: MsFlags, flag_str: &str) -> Option<MsFlags> {
+    match flag_str {
+        "async" => flags &= !MsFlags::MS_SYNCHRONOUS,
+        "atime" => flags &= !MsFlags::MS_NOATIME,
+        "dev" => flags &= !MsFlags::MS_NODEV,
+        "diratime" => flags &= !MsFlags::MS_NODIRATIME,
+        "exec" => flags &= !MsFlags::MS_NOEXEC,
+        "nomand" => flags &= !MsFlags::MS_MANDLOCK,
+        "norelatime" => flags &= !MsFlags::MS_RELATIME,
+        "nostrictatime" => flags &= !MsFlags::MS_STRICTATIME,
+        "rw" => flags &= !MsFlags::MS_RDONLY,
+        "suid" => flags &= !MsFlags::MS_NOSUID,
+        "bind" => flags |= MsFlags::MS_BIND,
```

Comment:
Nit: Unsorted list.

Reply:
The flags have been divided into two groups: clearing flags and setting flags. Within each group, there are ordered. Add a comment to split these two groups.